### PR TITLE
Migration guide for #10

### PR DESCRIPTION
**Pull request by Ericson2314**

### CODE REVIEW

File: guides/functor-combinator-instances-and-class1s.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+However, the breakage is of limited scope.
+An impact analysis (please see [comments following here](https://github.com/haskell/core-libraries-committee/issues/10#issuecomment-1166669613))
+showed that a only handle of packages that sucessfully pass dependency resolution on Stackage `nightly-2022-06-17` break with this change.
+Patches have been submitted to each of them.
+
+The migration policy for this change is backward-compatible: you can migrate already and still retain compatibility with existing GHCs.
+Because of this, CLC suggests applying patches at your earliest convenience.
+
+1. Add any missing instances
```
Comment (Bodigrim) [inline guides/functor-combinator-instances-and-class1s.md:43]: Let's spell this out more explicitly "If you defined `instance Eq1 f` but omitted `instance Eq a => Eq (f a)`, you must now define the latter, e. g., as `(==) = eq1`. If you defined `instance Eq2` f ..."

Please also link relevant PRs from https://github.com/haskell/core-libraries-committee/issues/10#issuecomment-1166669613 as examples.

File: guides/functor-combinator-instances-and-class1s.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# What
+
+CLC has approved the [proposal #10](https://github.com/haskell/core-libraries-committee/issues/10), to relax instances for Functor combinators and put superclasses on \<class>1 to make the change less-breaking.
```
Comment (Bodigrim) [inline guides/functor-combinator-instances-and-class1s.md:3]: Please put an explicit list of classes affected.

File: guides/functor-combinator-instances-and-class1s.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+This is a breaking change:
+
+1. People could define e.g. `Eq1` without `Eq`, or `Eq2` without `Eq1`
+
+2. People could write complicated instances on various abstract machinary that doesn't abide by the new superclasses.
+
+However, the breakage is of limited scope.
+An impact analysis (please see [comments following here](https://github.com/haskell/core-libraries-committee/issues/10#issuecomment-1166669613))
+showed that a only handle of packages that sucessfully pass dependency resolution on Stackage `nightly-2022-06-17` break with this change.
```
Comment (konsumlamm) [inline guides/functor-combinator-instances-and-class1s.md:37]:
```suggestion
showed that only a few of the packages that sucessfully pass dependency resolution on Stackage `nightly-2022-06-17` break with this change.
```

File: guides/functor-combinator-instances-and-class1s.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+   As one can see, this comes from rather fancy stuff.
+
+   `streaming` could be simpler, as the pre-existing comment points out, by just requiring a `Show1 f` constraint.
+   The other instance that required such a constraint did not need any migration.
+
+   `text-show` had `*1` instances that also converted between `String`- and `Text`-based showing.
+   This violates the superclass.
+   The instance must instead "pass through" `String`-based shows to `String`-based shows.
+
+Note what is *not* a breaking change is the changing of the instances for `Compose` etc.
```
Comment (konsumlamm) [inline guides/functor-combinator-instances-and-class1s.md:59]:
```suggestion
Note that changing the instances for `Compose` etc. is *not* a breaking change.
```

File: guides/functor-combinator-instances-and-class1s.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+Thanks to the new superclasses, the old instance constraints imply the new instance constraints, so existing generic code using those types will continue to work.
+`Compose` and friends are much more common than the `*1` and `*2` classes, so this yields far less breakage over all.
```
Comment (konsumlamm) [inline guides/functor-combinator-instances-and-class1s.md:61]:
```suggestion
`Compose` and friends are much more common than the `*1` and `*2` classes, so this yields far less breakage overall.
```

File: guides/functor-combinator-instances-and-class1s.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+1. People could define e.g. `Eq1` without `Eq`, or `Eq2` without `Eq1`
+
+2. People could write complicated instances on various abstract machinary that doesn't abide by the new superclasses.
+
+However, the breakage is of limited scope.
+An impact analysis (please see [comments following here](https://github.com/haskell/core-libraries-committee/issues/10#issuecomment-1166669613))
+showed that a only handle of packages that sucessfully pass dependency resolution on Stackage `nightly-2022-06-17` break with this change.
+Patches have been submitted to each of them.
+
+The migration policy for this change is backward-compatible: you can migrate already and still retain compatibility with existing GHCs.
```
Comment (konsumlamm) [inline guides/functor-combinator-instances-and-class1s.md:40]:
```suggestion
The migration policy for this change is backwards-compatible: you can migrate already and still retain compatibility with existing GHCs.
```

File: guides/functor-combinator-instances-and-class1s.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+At the very earliest, the change may appear in GHC 9.6 (~ Q1 2023).
+
+# How
+
+This is a breaking change:
+
+1. People could define e.g. `Eq1` without `Eq`, or `Eq2` without `Eq1`
+
+2. People could write complicated instances on various abstract machinary that doesn't abide by the new superclasses.
```
Comment (mixphix) [inline guides/functor-combinator-instances-and-class1s.md:33]:
```suggestion
2. People can no longer write complicated instances on various abstract machinery that doesn't abide by the new superclasses.
```

File: guides/functor-combinator-instances-and-class1s.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+# When
+
+At the very earliest, the change may appear in GHC 9.6 (~ Q1 2023).
+
+# How
+
+This is a breaking change:
+
+1. People could define e.g. `Eq1` without `Eq`, or `Eq2` without `Eq1`
```
Comment (mixphix) [inline guides/functor-combinator-instances-and-class1s.md:31]:
```suggestion
1. People can no longer define e.g. `Eq1` without `Eq`, or `Eq2` without `Eq1`
```

**Review by mixphix — CHANGES_REQUESTED** [review]
I think this wording makes it clearer which behaviour is the case now.

**Comment by Bodigrim** [comment]
@Ericson2314 could you please apply the suggestions? I'd really love to finish up with this proposal, it's almost a year old ;)

File: guides/functor-combinator-instances-and-class1s.md (hunk 2 above)
Comment (Ericson2314) [reply guides/functor-combinator-instances-and-class1s.md]: Done

File: guides/functor-combinator-instances-and-class1s.md (hunk 4 above)
Comment (Ericson2314) [reply guides/functor-combinator-instances-and-class1s.md:81]:
```suggestion
Note that in contrast to the above, the other half of proposal 10 are *not* breaking changes.
In conjunction with the changes to the classes, the changes to instances for `Compose` and other such data types is backwards compatible..
```

**Comment by Ericson2314** [comment]
OK @Bodigrim I think I (belatedly) got everything. Let me know if not!

**Review by Bodigrim — APPROVED** [review]

**Comment by Bodigrim** [comment]
@mixphix @konsumlamm are you happy with the changes?

**Comment by mixphix** [comment]
LGTM

File: guides/functor-combinator-instances-and-class1s.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
+# What
+
+CLC has approved the [proposal #10](https://github.com/haskell/core-libraries-committee/issues/10), to relax instances for Functor combinators and put superclasses on \<class>1 and \<class2> to make the change less-breaking.
```
Comment (konsumlamm) [inline guides/functor-combinator-instances-and-class1s.md:3]:
```suggestion
CLC has approved the [proposal #10](https://github.com/haskell/core-libraries-committee/issues/10), to relax instances for Functor combinators and put superclasses on \<class>1 and \<class>2 to make the change less-breaking.
```

File: guides/functor-combinator-instances-and-class1s.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+The effected classes are:
```
Comment (konsumlamm) [inline guides/functor-combinator-instances-and-class1s.md:4]:
```suggestion
The affected classes are:
```

File: guides/functor-combinator-instances-and-class1s.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+# When
+
+At the very earliest, the change may appear in GHC 9.6 (~ Q1 2023).
+
+# How
+
+This is a breaking change:
+
+1. People can no longer define e.g. `Eq1` without `Eq`, or `Eq2` without `Eq1`
```
Comment (konsumlamm) [inline guides/functor-combinator-instances-and-class1s.md:37]:
```suggestion
1. People can no longer define e.g. `Eq1` without `Eq`, or `Eq2` without `Eq1`.
```

**Comment by Bodigrim** [comment]
Thanks @Ericson2314!

**Pull request merged**